diff --git a/renderer/include/rive/renderer/metal/render_context_metal_impl.h b/renderer/include/rive/renderer/metal/render_context_metal_impl.h
index fee553c0..49cca4d0 100644
--- a/renderer/include/rive/renderer/metal/render_context_metal_impl.h
+++ b/renderer/include/rive/renderer/metal/render_context_metal_impl.h
@@ -133,6 +133,8 @@ public:
                                   uint32_t mipLevelCount,
                                   const uint8_t imageDataRGBAPremul[]) override;
 
+    rcp<Texture> makeImageTexture(uint32_t width, uint32_t height, id<MTLTexture> texture);
+
     rcp<RenderCanvas> makeRenderCanvas(uint32_t width,
                                        uint32_t height) override;
 
diff --git a/renderer/src/metal/render_context_metal_impl.mm b/renderer/src/metal/render_context_metal_impl.mm
index ba7a51f5..9d735d58 100644
--- a/renderer/src/metal/render_context_metal_impl.mm
+++ b/renderer/src/metal/render_context_metal_impl.mm
@@ -868,6 +868,13 @@ void ensureMipmaps(id<MTLCommandBuffer> commandBuffer) const
         m_gpu, width, height, mipLevelCount, imageDataRGBAPremul);
 }
 
+rcp<Texture> RenderContextMetalImpl::makeImageTexture(uint32_t width,
+                                                      uint32_t height,
+                                                      id<MTLTexture> texture)
+{
+    return make_rcp<TextureMetalImpl>(texture, width, height);
+}
+
 rcp<RenderCanvas> RenderContextMetalImpl::makeRenderCanvas(uint32_t width,
                                                            uint32_t height)
 {
